### PR TITLE
Add coverage for py-mpitests.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by multiphaseCFD** [comment]
Thanks @vincentmr ! Would you like to add failing cases for the expval calculation Hermitian obs?

**Comment by vincentmr** [comment]
> Thanks @vincentmr ! Would you like to add failing cases for the expval calculation Hermitian obs?

No more failing cases after I `Bcast`ed `U` and `init_state`. I just pushed a new commit.

File: .github/workflows/tests_linux_x86_mpi.yml [hunk 1]
```diff
@@ -252,15 +248,75 @@ jobs:
...
+          if-no-files-found: error
+
+      - name: Cleanup
+        if: always()
+        run: |
+          rm -rf ${{ steps.setup_venv.outputs.venv_name }}
+          rm -rf * .git .gitignore .github
+          pip cache purge
+
+  upload-to-codecov-linux-cpp:
```
Comment (multiphaseCFD) [inline .github/workflows/tests_linux_x86_mpi.yml:270]: will the results of `openmpi` test overrid the `mpich` ones?

Comment (vincentmr) [reply .github/workflows/tests_linux_x86_mpi.yml:270]: Good question, maybe the last one is the one to report? Is that important?

Comment (multiphaseCFD) [reply .github/workflows/tests_linux_x86_mpi.yml:270]: yea, `codecov` would complain that there is lack of coverage for `mpich` or `openmpi`.

**Comment by multiphaseCFD** [comment]
Could you also summarize it in changelog? Thanks

**Pull request merged**